File: code/appd-12.ex
l = [1,2,3]
hd(l) # -> 1
tl(l) # -> [2,3]
length(l) # -> 3
t = {hd(l),tl(l),"hello world"} # -> {1, [2, 3], "hello world"}
tuple_size(t) # -> 3
elem(t,1) # -> [2,3]
put_elem(t,1,[0]) # -> {1, [0], "hello world"}
t # t unchanged -> {1, [2, 3], "hello world"}

File: code/appd-12.erl
L = [1,2,3].
hd(L). % -> 1
tl(L). % -> [2,3]
length(L). % -> 3
T = {hd(L),tl(L),"hello world"}. % -> {1,[2,3],"hello world"}
tuple_size(T). % -> 3
element(2,T). % -> [2,3]
setelement(2,T,[0]). % -> {1,[0],"hello world"}
T. % T unchanged -> {1,[2,3],"hello world"}
